Sources/BrowserServicesKit/Autofill/Matchers/AutofillUrlMatcher.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  AutofillUrlMatcher.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+
+public protocol AutofillUrlMatcher {
+    func cleanRawUrl(_ rawUrl: String) -> String
+    func isMatchingForAutofill(currentSite: String, savedSite: String, tld: TLD) -> Bool
+    func normalizeSchemeForAutofill(_ rawUrl: String) -> URLComponents?
+}
+
+public struct AutofillDomainNameUrlMatcher: AutofillUrlMatcher {
+
+    public init() {}
+
+    public func cleanRawUrl(_ rawUrl: String) -> String {
```

Comment:
Nit: It's not immediately obvious to me what a raw url is, and if it's feasible to expand on what cleaning entails in the function name I think that would be good